### ISSUE TITLE
Doesn't answer after push and then ANRs when trying to open

### ISSUE THREAD

**Issue by brianjmurrell** [BelledonneCommunications/linphone-android#1125]
Using the kotlin_rewrite branch, I am finding that Linphone is not answering after receiving a push.  If I then try to open it, to collect a log, for example, the application actually triggers an ANR.  I choose close app and them am able to open it.

The push was sent at 2020-08-16 15:51:54 and the confirmation:

```json
{
  "multicast_id":[redacted,"success":1,
  "failure":0,
  "canonical_ids":0,
  "results":[{"message_id":"0:[redacted]%[redacted]"}]
}
```

and the [linphone log](http://brian.interlinx.bc.ca//linphone-NA-then-ANR.txt) from a few minutes before the call and push until I opened linphone and sent a log.

**Comment by Viish** [comment]
From your logs it seems the app crashes in the registration_state_changed callback.
Check your adb logcats for a stack trace.
Also there is no trace either of a ANR, please attach the /data/anr/traces.txt file from your device.

**Comment by Viish** [comment]
Ok I know what happen.
I checked the SDK version you use and you don't have the latest version but a version from 3 weeks that had an endless loop issue.
Update your SDK and you will be fine.

**Comment by brianjmurrell** [comment]
Once again, I have to wonder why the project lets one build with old/broken/SDKs.

Isn't there some way to stop/break the build if some minimum SDK version is not met, or even more ideally, update the SDK automatically?

What is the canonical way of updating the SDK without wiping out the entire world (or at least the entire gradle cache?

**Comment by Viish** [comment]
There was a  2 weeks time frame during which SDK with this issue were uploaded, and yes we didn't remove the uploaded binary from the maven as new version was available with the fix.
Your Android Studio should take at most 48h before upgrading the SDK to a newer version if available unless you instructed it to do otherwise.
Also as it is mentionned, the master is in alpha state which can occasionally have this kind of issue. Stick to releases if you want a stable version.

**Comment by brianjmurrell** [comment]
So what is the minimum SDK version that I should be using?  I will do a new `./gradlew installDebug` and see if I get a newer version as I don't believe I have instructed otherwise:

```
    implementation 'org.linphone:linphone-sdk-android:4.5+'
```

from `dependencies { }` in `app/build.gradle`.

**Comment by Viish** [comment]
For a stable version you should use `org.linphone:linphone-sdk-android:4.4+` but it isn't compatible with kotlin_rewrite branch.
If you want to stick on kotlin_rewrite use latest 4.5: `4.5.0-alpha.64+4fde8b4`

**Comment by brianjmurrell** [comment]
Indeed, `./gradlew installDebug` did update the SDK and at least after an initial test, it seems that the issue here is resolved.  I will give it some more testing throughout the day.

But yeah, since I am using the kotlin_rewrite branch, I do need the 4.5 SDK.

Is the kotlin_rewrite->master update behind schedule?  I thought I had an impression somewhere that this was going to land June-ish time frame.  Was I mistaken about that?

**Comment by Viish** [comment]
Actually kotlin_rewrite will be come next master for 5.0 release (SDK & app) that was indeed planned for this summer.
However we need to do a 4.5 release first of the SDK and we are indeed behind schedule.
Once 4.5 will be released we will start 5.0 release process and only then we will merge the kotlin branch into master.
